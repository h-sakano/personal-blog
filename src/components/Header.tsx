import { Link } from 'gatsby';
import React from 'react';
import { Typography } from 'antd';
import styles from './Header.module.css';

interface HeaderProps {
  description?: string;
  siteTitle?: string;
}

const Header: React.FC<HeaderProps> = ({ description, siteTitle }) => (
  <header className={styles.header}>
    <Typography.Title>
      <Link to="/" className={styles.link}>
        {siteTitle}
      </Link>
    </Typography.Title>
    <Typography.Text>{description}</Typography.Text>
  </header>
);

export default Header;
