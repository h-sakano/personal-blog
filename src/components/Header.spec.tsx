import { render } from '@testing-library/react';
import * as React from 'react';
import Header from './Header';

describe('<Header />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Header description="概要" siteTitle="タイトル" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
