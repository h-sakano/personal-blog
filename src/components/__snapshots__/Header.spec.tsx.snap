// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<DocumentFragment>
  <header
    class="text-center"
  >
    <h1>
      <a
        class="text-4xl text-current"
        href="/"
      >
        タイトル
      </a>
    </h1>
    <div
      class="mt-2 text-gray-600"
    >
      概要
    </div>
  </header>
</DocumentFragment>
`;
